common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala

```diff
@@ -20,10 +20,10 @@ sealed trait Work[State <: WorkState] {
   def identifiers: List[SourceIdentifier] =
     sourceIdentifier :: data.otherIdentifiers
 
-  def transition[OutState <: WorkState](args: OutState#TransitionArgs)(
+  def transition[OutState <: WorkState](args: OutState#TransitionArgs = ())(
```

Comment:
Really baffled how this compiles :thinking: `TranstionArgs` isn't necessarilly of type `Unit` so confused how using that as a default works here

Reply:
It catches this when you apply it to a type which isn't `Unit`:

```
[E]      type mismatch;
[E]       found   : Unit
[E]       required: uk.ac.wellcome.models.work.internal.WorkState.Denormalised#TransitionArgs
[E]          (which expands to)  uk.ac.wellcome.models.work.internal.Relations[uk.ac.wellcome.models.work.internal.DataState.Unidentified]
[E]      Error occurred in an application involving default arguments.
[E]      L251:     work.transition[Denormalised]()
[E]                               ^
```

Reply:
I was going to comment that on it haha - part of me wonders if it will compile in CI and it's just that my IDE hasn't noticed it...

Reply:
No I checked it out locally and it works!